### PR TITLE
KARAF-7554 - add feature:status command

### PR DESCRIPTION
**Pull request by awrb**
It's a subtask from [KARAF-5193](https://issues.apache.org/jira/browse/KARAF-5193), which is pretty old but I think it's still relevant (at least feature:status was never implemented).

### CODE REVIEW

File: features/command/src/main/java/org/apache/karaf/features/command/completers/FeatureVersionCompleter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public int complete(Session session, CommandLine commandLine, List<String> candidates) {
+        StringsCompleter delegate = new StringsCompleter();
+        String[] args = commandLine.getArguments();
+        // args look like this at this point: [feature:status, wrapper, '']
+        if (args.length >= 3) {
+            String featureArg = args[1];
+            try {
+                Feature[] features = featuresService.getFeatures(featureArg);
+                List<String> versions = Arrays.stream(features).map(Feature::getVersion).collect(Collectors.toList());
+                delegate.getStrings().addAll(versions);
```
Comment (awrb) [inline features/command/src/main/java/org/apache/karaf/features/command/completers/FeatureVersionCompleter.java:32]: I wanted the completer to work if there are multiple versions of a single feature available, but sadly `versions` is always a singleton here. In `FeatureReq` the latest version is returned:
```
    public Stream<Feature> getMatchingFeatures(Map<String, Map<String, Feature>> allFeatures) {
        Pattern pattern = Pattern.compile(name);
        Function<String, Optional<Feature>> func = featureName -> {
            Feature matchingFeature = null;
            if (pattern.matcher(featureName).matches()) {
                Map<String, Feature> versions = allFeatures.get(featureName);
                matchingFeature = getLatestFeature(versions, versionRange);
            }
            return Optional.ofNullable(matchingFeature);
        };
        return allFeatures.keySet().stream().map(func).filter(Optional::isPresent).map(Optional::get);
    }
```
It does not seem possible to get all versions via `FeaturesService`

**Review by oscerd — APPROVED** [review]

File: features/command/src/main/java/org/apache/karaf/features/command/completers/FeatureVersionCompleter.java (hunk 1 above)
Comment (awrb) [reply features/command/src/main/java/org/apache/karaf/features/command/completers/FeatureVersionCompleter.java:48]:
```
 private List<String> getAllVersionsOfFeature(String feature, FeaturesService featuresService) throws Exception {
        List<String> versions = new ArrayList<>();
        for (Repository repo : featuresService.listRepositories()) {
            for (Feature f : repo.getFeatures()) {
                if (f.getName().equals(feature)) {
                    versions.add(f.getVersion());
                }
            }
        }

        return versions;
    }
```

**Comment by awrb** [comment]
The pipeline failed but this probably wasn't caused by my changes? Is this a known instability?
I see something similar here: https://github.com/apache/karaf/runs/8186042837

**Comment by jbonofre** [comment]
Yes there are couple of flaky tests.

**Review by jbonofre — APPROVED** [review]
Maybe worth (later) to add a `status` operation on the FeatureMBean.

**Pull request merged**